cpp/src/column/column_view.cpp

```diff
@@ -124,11 +126,9 @@ mutable_column_view::operator column_view() const
 
 size_type count_descendants(column_view parent)
 {
-  size_type count{parent.num_children()};
-  for (size_type i = 0; i < parent.num_children(); ++i) {
-    count += count_descendants(parent.child(i));
-  }
-  return count;
+  auto op    = [&](auto i) { return count_descendants(parent.child(i)); };
+  auto begin = cudf::test::make_counting_transform_iterator(0, op);
+  return std::accumulate(begin, begin + parent.num_children(), size_type{parent.num_children()});
```

Comment:
I think you can do one better by just using the `child_begin/child_end` iterators: https://github.com/rapidsai/cudf/blob/c6dfa6e3ed8cf898faed244146c4da7ffd5d7cc7/cpp/include/cudf/column/column_view.hpp#L346

Reply:
If you use child iterators, seems you would not need a counting iterator anymore, so this would just use `make_transform_iterator` -- (minus one for `make_counting_transform_iterator` in cudf. :) )